### PR TITLE
[#0] Add default modifier tests

### PR DESCRIPTION
**Pull request by flamion**
I just wanted to add a few more tests, it didn't seem like we had enough :point_right: :point_left: 
image

### CODE REVIEW

File: backend/shork/src/test/kotlin/compiler/TestParser.kt [hunk 1]
```diff
@@ -614,5 +622,353 @@ internal class TestParser {
...
+                    Triple(TokenType.JMP, Jmp::class, Modifier.B),
+                    Triple(TokenType.JMZ, Jmz::class, Modifier.B),
+                    Triple(TokenType.JMN, Jmn::class, Modifier.B),
+                    Triple(TokenType.DJN, Djn::class, Modifier.B),
+                    Triple(TokenType.SPL, Spl::class, Modifier.B),
+                )
+
+            for (instruction in instructionWithModifier) {
+                for (firstAddressMode in addressModes) {
+                    for (secondAddressMode in addressModes) {
```
Comment (nikkischnelle) [inline backend/shork/src/test/kotlin/compiler/TestParser.kt:657]: at this point, we might want to think about a `cartesianProduct(list1, list2, list3)` helper or something lol

File: backend/shork/src/test/kotlin/compiler/TestParser.kt [hunk 2]
```diff
@@ -614,5 +622,353 @@ internal class TestParser {
...
+                        inputWithExpected.add(Pair(tokens, expectedInstance))
+                    }
+                }
+            }
+
+            return inputWithExpected.map { Arguments.of(it.first, it.second) }
+        }
+
+        // MOV, SEQ, SNE, CMP, ADD, SUB, MUL, DIV, MOD, SLT, LDP, STP
+        private fun `provide A Mode immediate, B whatever then AB`(): List<Arguments> {
```
Comment (PietHelzel) [inline backend/shork/src/test/kotlin/compiler/TestParser.kt:710]: The "then AB" at the end is a bit confusing to read. Further down you changed it to "then Modifier is ..." in some function names, I would prefer that for clarity.

File: backend/shork/src/test/kotlin/compiler/TestParser.kt [hunk 3]
```diff
@@ -614,5 +622,353 @@ internal class TestParser {
...
+
+                    inputWithExpected.add(Pair(tokens, expectedInstance))
+                }
+            }
+
+            return inputWithExpected.map { Arguments.of(it.first, it.second) }
+        }
+
+        // MOV, SEQ, SNE, CMP, ADD, SUB, MUL, DIV, MOD, SLT, LDP, STP
+        private fun `provide B-Mode immediate, A-Mode whatever then B`(): List<Arguments> {
```
Comment (PietHelzel) [inline backend/shork/src/test/kotlin/compiler/TestParser.kt:771]: Same as above

**Review by PietHelzel — CHANGES_REQUESTED** [review]

File: backend/shork/src/test/kotlin/compiler/TestParser.kt (hunk 1 above)
Comment (flamion) [reply backend/shork/src/test/kotlin/compiler/TestParser.kt:657]: Yeah, something like that might be nice

**Review by PietHelzel — APPROVED** [review]

**Pull request merged**